src/modules/flow/json/json.c

```diff
@@ -224,4 +243,226 @@ json_object_get_key_in_process(struct sol_flow_node *node, void *data, uint16_t
     return json_object_key_process(node, mdata);
 }
 
+static int
+json_object_length_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_blob *in_value;
+    struct sol_json_scanner scanner;
+    enum sol_json_loop_reason reason;
+    struct sol_json_token token, key, value;
+    struct sol_irange len = { 0, 0, INT32_MAX, 1 };
+
+    r = sol_flow_packet_get_json_object(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_json_scanner_init(&scanner, in_value->mem, in_value->size);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        if (len.val == INT32_MAX)
+            return -ERANGE;
+
+        len.val++;
+    }
+
+    return sol_flow_send_irange_packet(node,
+        SOL_FLOW_NODE_TYPE_JSON_OBJECT_LENGTH__OUT__OUT, &len);
+}
+
+static int
+json_object_get_all_keys_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_blob *in_value;
+    struct sol_json_scanner scanner;
+    enum sol_json_loop_reason reason;
+    struct sol_json_token token, key, value;
+    struct sol_buffer buffer;
+    bool empty = true;
+
+    r = sol_flow_packet_get_json_object(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_json_scanner_init(&scanner, in_value->mem, in_value->size);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        r = sol_json_token_get_unescaped_string(&key, &buffer);
+        SOL_INT_CHECK(r, < 0, r);
+        r = sol_flow_send_string_slice_packet(node,
+            SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_ALL_KEYS__OUT__OUT,
+            sol_buffer_get_slice(&buffer));
+        sol_buffer_fini(&buffer);
+        SOL_INT_CHECK(r, < 0, r);
+
+        empty = false;
+    }
+
+    return sol_flow_send_boolean_packet(node,
+        SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_ALL_KEYS__OUT__EMPTY, empty);
+}
+
+struct sol_json_array_index {
+    struct sol_blob *json_array;
+    int32_t index;
+};
+
+
+static bool
+json_array_get_at_index(struct sol_json_scanner *scanner, struct sol_json_token *token, int32_t index, enum sol_json_loop_reason *reason)
+{
+    int32_t cur_index = 0;
+
+    SOL_JSON_SCANNER_ARRAY_LOOP_ALL(scanner, token, *reason) {
+        if (index == cur_index)
```

Comment:
avoid using `index` as it shadows a `<strings.h>` function of that name. New compilers are savvy enough to avoid such errors, but old tools will trigger warnings due shadow declaration.